benchmarks/confbayes/example_2moons_CB.py

```diff
@@ -0,0 +1,113 @@
+import jax.numpy as jnp
+import numpy as np
+from sklearn.datasets import make_moons
+
+from fortuna.data import (
+    DataLoader,
+    InputsLoader,
+)
+from fortuna.metric.classification import accuracy
+from fortuna.model import MLP
+from fortuna.prob_model import (
+    ADVIPosteriorApproximator,
+    CalibConfig,
+    CalibMonitor,
+    FitConfig,
+    FitMonitor,
+    FitOptimizer,
+    ProbClassifier,
+)
+
+# Approximate Conformal Bayes: 2 Moons Example
+
+## Simulate data
+n = 1000
+train_data = make_moons(n_samples=n, noise=0.2, random_state=0)
+n_test = 500
+test_data = make_moons(n_samples=n_test, noise=0.2, random_state=1)
+
+import flax.linen as nn
+import optax
+
+## Fit Flax models
+# Setup DataLoaders
+train_data_loader = DataLoader.from_array_data(
+    train_data, batch_size=128, shuffle=True, prefetch=True
+)
+
+test_inputs_loader = InputsLoader.from_array_inputs(
+    test_data[0], batch_size=128, prefetch=True
+)
+
+# Setup Flax Model
+output_dim = 2
+prob_model = ProbClassifier(
+    model=MLP(output_dim=output_dim, activations=(nn.tanh, nn.tanh)),
+    posterior_approximator=ADVIPosteriorApproximator(),
+)
+
+# Train model
+status = prob_model.train(
+    train_data_loader=train_data_loader,  # Remove validation set to make it simpler
+    fit_config=FitConfig(
+        monitor=FitMonitor(metrics=(accuracy,), early_stopping_patience=10),
+        optimizer=FitOptimizer(method=optax.adam(1e-1)),
+    ),
+    calib_config=CalibConfig(monitor=CalibMonitor(early_stopping_patience=2)),
+)
+
+### Run conformal Bayes
+error = 0.2
+conf_set, ESS = prob_model.predictive.conformal_set(
+    train_data_loader,
+    test_inputs_loader,
+    error=error,
+    n_posterior_samples=100,
+    ESS=True,
+)
+
+
+### Evaluate conformal Bayes
+# Evaluate conformal method
+cov = np.zeros(n_test)
+length = np.zeros(n_test)
+y_test = test_data[1]  # test data points
+
+# Loop through test data points and compute conformal sets for each
+for j in range(n_test):
```

Comment:
Maybe we can rather use a `vmap` here. A standard for loop is pretty slow if `n_test` is large.